src/plugins/dashboard/public/locator.ts

```diff
@@ -116,26 +128,27 @@ export class DashboardAppLocatorDefinition implements LocatorDefinition<Dashboar
       }
     };
 
+    const { filters, panels, ...restParams } = params;
+    const state: ForwardedDashboardState = restParams;
+
     // leave filters `undefined` if no filters was applied
     // in this case dashboard will restore saved filters on its own
-    const filters = params.filters && [
+    state.filters = params.filters && [
       ...(await getSavedFiltersFromDestinationDashboardIfNeeded()),
       ...params.filters,
     ];
 
-    let path = setStateToKbnUrl(
-      '_a',
-      cleanEmptyKeys({
-        query: params.query,
-        filters: filters?.filter((f) => !esFilters.isFilterPinned(f)),
-        viewMode: params.viewMode,
-        panels: params.panels,
-        savedQuery: params.savedQuery,
-      }),
-      { useHash },
```

Comment:
what was `useHash` used for ?

Reply:
This was used to dictate how app state is stored, on the URL or in sessionStorage. Since we no longer generate the URL from here with `_a` this `setStateToKbnUrl` step was deleted entirely.